src/mem/AllocatorProxy.h

```diff
@@ -9,7 +9,10 @@
 #ifndef _SQUID_SRC_MEM_ALLOCATORPROXY_H
 #define _SQUID_SRC_MEM_ALLOCATORPROXY_H
 
-class MemAllocator;
+// XXX: remove AllocatorProxy.h include from mem/forward.h
```

Comment:
Any reason this XXX should not be addressed in this PR? It seems to be very much in scope, but I am probably missing some caveats.

Reply:
I have scoped this PR at only the MemAllocator class. Mem::AllocatorProxy is a very widely used class and would drag in a lot of include sequence changes that bloat the PR.